migtests/scripts/functions.sh

```diff
@@ -30,28 +30,52 @@ print_env() {
 run_psql() {
 	db_name=$1
 	sql=$2
-	psql "postgresql://${SOURCE_DB_USER}:${SOURCE_DB_PASSWORD}@${SOURCE_DB_HOST}:${SOURCE_DB_PORT}/${db_name}" -c "${sql}"
+	psql "postgresql://${SOURCE_DB_ADMIN_USER}:${SOURCE_DB_ADMIN_PASSWORD}@${SOURCE_DB_HOST}:${SOURCE_DB_PORT}/${db_name}" -c "${sql}"
+}
+
+grant_user_permission_postgresql() {
+	db_name=$1
+
+	echo "SELECT 'GRANT USAGE ON SCHEMA ' || schema_name || ' TO ${SOURCE_DB_USER};' FROM information_schema.schemata; \gexec" | psql "postgresql://postgres:secret@${SOURCE_DB_HOST}:${SOURCE_DB_PORT}/${db_name}" 
+	echo "SELECT 'GRANT SELECT ON ALL TABLES IN SCHEMA ' || schema_name || ' TO ${SOURCE_DB_USER};' FROM information_schema.schemata; \gexec" | psql "postgresql://postgres:secret@${SOURCE_DB_HOST}:${SOURCE_DB_PORT}/${db_name}" 
+	echo "SELECT 'GRANT SELECT ON ALL SEQUENCES IN SCHEMA ' || schema_name || ' TO ${SOURCE_DB_USER};' FROM information_schema.schemata; \gexec" | psql "postgresql://postgres:secret@${SOURCE_DB_HOST}:${SOURCE_DB_PORT}/${db_name}" 
 }
 
 run_pg_restore() {
 	db_name=$1
 	file_name=$2
-	export PGPASSWORD=${SOURCE_DB_PASSWORD}
+	export PGPASSWORD=${SOURCE_DB_ADMIN_PASSWORD}
 	pg_restore --no-password -h ${SOURCE_DB_HOST} -p ${SOURCE_DB_PORT} \
-		-U ${SOURCE_DB_USER} -d ${db_name} ${file_name}
+		-U ${SOURCE_DB_ADMIN_USER} -d ${db_name} ${file_name}
 	unset PGPASSWORD
 }
 
 run_ysql() {
 	db_name=$1
 	sql=$2
-	psql "postgresql://yugabyte:${TARGET_DB_PASSWORD}@${TARGET_DB_HOST}:${TARGET_DB_PORT}/${db_name}" -c "${sql}"
+	psql "postgresql://${TARGET_DB_ADMIN_USER}:${TARGET_DB_ADMIN_PASSWORD}@${TARGET_DB_HOST}:${TARGET_DB_PORT}/${db_name}" -c "${sql}"
 }
 
 run_mysql() {
 	db_name=$1
 	sql=$2
-	mysql -u ${SOURCE_DB_USER} -p${SOURCE_DB_PASSWORD} -h ${SOURCE_DB_HOST} -P ${SOURCE_DB_PORT} -D ${db_name} -e "${sql}"
+	mysql -u ${SOURCE_DB_ADMIN_USER} -p${SOURCE_DB_ADMIN_PASSWORD} -h ${SOURCE_DB_HOST} -P ${SOURCE_DB_PORT} -D ${db_name} -e "${sql}"
+}
+
+grant_user_permission_mysql() {
+	db_name=$1
+	
+	run_mysql ${db_name} "GRANT PROCESS ON *.* TO '${SOURCE_DB_USER}'@'${SOURCE_DB_HOST}';"
+	run_mysql ${db_name} "GRANT SELECT ON ${SOURCE_DB_NAME}.* TO '${SOURCE_DB_USER}'@'${SOURCE_DB_HOST}';"
+	run_mysql ${db_name} "GRANT SHOW VIEW ON ${SOURCE_DB_NAME}.* TO '${SOURCE_DB_USER}'@'${SOURCE_DB_HOST}';"
+	run_mysql ${db_name} "GRANT TRIGGER ON ${SOURCE_DB_NAME}.* TO '${SOURCE_DB_USER}'@'${SOURCE_DB_HOST}';"
+
+	# For MySQL >= 8.0.20
+	# run_mysql ${db_name} "GRANT SHOW_ROUTINE  ON *.* TO 'ybvoyager'@'${SOURCE_DB_HOST}';"
+
+	# For older versions
+	run_mysql ${db_name} "GRANT SELECT ON *.* TO '${SOURCE_DB_USER}'@'${SOURCE_DB_HOST}';"
```

Comment:
Use bash array syntax (https://www.freecodecamp.org/news/bash-array-how-to-declare-an-array-of-strings-in-a-bash-script/) to simplify this block. The code structure should be similar to the following Python syntax:

```
cmds = [
  "GRANT PROCESS ON *.*",
  "GRANT SELECT ON ${SOURCE_DB_NAME}.*",
  ...
]
for cmd in cmds:
  full_cmd = "%s TO '%s'@'%s';" % (cmd, source_db_user, source_db_host)
  run_mysql(db_name, full_cmd)
```